v1/src/main/java/com/google/cloud/teleport/datadog/DatadogEventWriter.java

```diff
@@ -443,6 +460,10 @@ public Builder withInputBatchCount(ValueProvider<Integer> inputBatchCount) {
       if (inputBatchCount != null && inputBatchCount.isAccessible()) {
         Integer batchCount = inputBatchCount.get();
         if (batchCount != null) {
+          checkArgument(
+              batchCount >= minBatchCount(),
+              "inputBatchCount must be greater than or equal to %s",
+              minBatchCount());
```

Comment:
`minBatchCount()` must not be `null` by the time this check is performed. That's why it's now part of the builder factory methods `newBuilder()` and `newBuilder(Integer minBatchCount)`